File: src/main/java/net/hellomouse/progression_change/registries/ProgressionModBlockEntityRegistry.java
package net.hellomouse.progression_change.registries;

import net.hellomouse.progression_change.ProgressionMod;
import net.hellomouse.progression_change.block.block_entity.BrickFurnaceBlockEntity;
import net.minecraft.block.entity.BlockEntityType;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;
import net.minecraftforge.registries.RegistryObject;

public class ProgressionModBlockEntityRegistry {
    public static final DeferredRegister<BlockEntityType<?>> DEF_REG = DeferredRegister.create(ForgeRegistries.BLOCK_ENTITY_TYPES, ProgressionMod.MODID);
    public static final RegistryObject<BlockEntityType<BrickFurnaceBlockEntity>> BRICK_FURNACE =
            DEF_REG.register("brick_furnace", () ->
                    BlockEntityType.Builder.create(BrickFurnaceBlockEntity::new, ProgressionModBlockRegistry.BRICK_FURNACE.get())
                            .build(null)
            );
}

File: src/main/kotlin/net/hellomouse/xeno_early_start/registries/ProgressionModBlockEntityRegistry.kt
package net.hellomouse.xeno_early_start.registries

import net.hellomouse.xeno_early_start.ProgressionMod
import net.hellomouse.xeno_early_start.block.block_entity.BrickFurnaceBlockEntity
import net.minecraft.block.BlockState
import net.minecraft.block.entity.BlockEntityType
import net.minecraft.block.entity.BlockEntityType.BlockEntityFactory
import net.minecraft.util.math.BlockPos
import net.minecraftforge.registries.DeferredRegister
import net.minecraftforge.registries.ForgeRegistries
import net.minecraftforge.registries.RegistryObject
import java.util.function.Supplier

object ProgressionModBlockEntityRegistry {
    val DEF_REG: DeferredRegister<BlockEntityType<*>> =
        DeferredRegister.create(ForgeRegistries.BLOCK_ENTITY_TYPES, ProgressionMod.Companion.MODID)

    @JvmField
    val BRICK_FURNACE: RegistryObject<BlockEntityType<BrickFurnaceBlockEntity>> =
        DEF_REG.register("brick_furnace", Supplier {
            BlockEntityType.Builder.create(BlockEntityFactory { pos: BlockPos, state: BlockState ->
                BrickFurnaceBlockEntity(
                    pos,
                    state
                )
            }, ProgressionModBlockRegistry.BRICK_FURNACE?.get())
                .build(null)
        }
        )
}
